platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/MerkleTreeSnapshotReader.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state;
+
+import static com.swirlds.common.io.streams.StreamDebugUtils.deserializeAndDebugOnFailure;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.io.streams.MerkleDataInputStream;
+import com.swirlds.platform.state.signed.SigSet;
+import com.swirlds.platform.state.snapshot.SignedStateFileUtils;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Set;
+
+/**
+ * Utility class for reading a snapshot of a {@link MerkleRoot} from disk.
+ */
+public class MerkleTreeSnapshotReader {
+
+    /**
+     * The previous version of the signed state file
+     */
+    public static final int INIT_STATE_FILE_VERSION = 1;
+    /**
+     * The current version of the signed state file. A file of this version no longer contains the signature set,
+     * instead the signature set is stored in a separate file.
+     */
+    public static final int SIG_SET_SEPARATE_STATE_FILE_VERSION = 2;
+    /**
+     * The supported versions of the signed state file
+     */
+    public static final Set<Integer> SUPPORTED_STATE_FILE_VERSIONS =
+            Set.of(INIT_STATE_FILE_VERSION, SIG_SET_SEPARATE_STATE_FILE_VERSION);
+    /**
+     * The signed state file was not versioned before, this byte was introduced to mark a versioned file
+     */
+    public static final byte VERSIONED_FILE_BYTE = Byte.MAX_VALUE;
+
+    /**
+     * This is a helper class to hold the data read from a state file.
+     * @param state the Merkle tree state
+     * @param hash the hash of the state
+     * @param sigSet the signature set
+     *
+     */
+    public record StateFileData(@NonNull MerkleRoot state, @NonNull Hash hash, @Nullable SigSet sigSet) {
+
+        @Override
+        @NonNull
+        public MerkleRoot state() {
+            return state;
+        }
+
+        @Override
+        @NonNull
+        public Hash hash() {
+            return hash;
+        }
+
+        @Override
+        @Nullable
+        public SigSet sigSet() {
+            return sigSet;
+        }
+    }
+
+    /**
+     * Reads a SignedState from disk using the provided snapshot reader function.
```

Comment:
This looks like a copy of an older javadoc because a reference to a "provided snapshot reader function" doesn't seem to make sense for this method. The only thing that is provided to it is a path. Could you please rephrase the javadoc?